src/simulator/privacy_tracker.py

```diff
@@ -42,6 +42,27 @@ class PrivacyBudget(NamedTuple):
     delta: float
 
 
+class SamplingBucketIndices(NamedTuple):
+    """Records the indices of sampling buckets.
+
+    For simplicity, we represent buckets as an interval [0, 1). A user is
+    mapped to a real number in this interval. A sampling bucket represents an
+    interval [smallest_index, smallest_index + sampling_rate). This means that
```

Comment:
As Matt mentioned below, wrap-arounds are possible.

I don't know what's the best way to clarify it here, perhaps just say [smallest_index, smallest_index + sampling_rate) mod [0, 1)...

Reply:
Changed as suggested now + added handling code & some test for it.